lib/history/sqlite/utils.go

```diff
@@ -0,0 +1,15 @@
+package sqlite
+
+import (
+	"github.com/mattn/go-sqlite3"
+)
+
+// isErrConstraintUnique returns true if error is a
+// sqlite3.ErrorConstraintUnique error.
+func isErrConstraintUnique(err error) bool {
+	sqliteErr, ok := err.(sqlite3.Error)
```

Comment:
Make sure the specified `err` has not been wrapped with `trace.Wrap` or do s safe `trace.Unwrap` to make sure you're always dealing with the original error.